guide/error_and_recovery.md

```diff
@@ -1,3 +1,61 @@
 # Error And Recovery
 
-*TODO*
+A key property of parsing unstructured data into a more structured, semantic form, is that that unstructured data may not be correctly arranged such that the conversion is possible. While this happens with (basically) any parser, different parsers might have different expectations of how much they want to do to recover from or provide info on malformed data. A high speed assembly interpreter might decide that it just needs to know that something bad happened, while a programming language compiler might want to ponder on the data and its surrounding context and provide as much information as possible to the end user, which might itself involve going over *more* data instead of failing immediately. Certain categories of errors may even be entirely recoverable from!
+
+## What is an [`Error`]?
+
+While Chumsky is responsible for the plumbing and detection of errors based on Parser Combinator outputs, you have full control over the creation, display, contained data and interactions for errors you define. Chumsky achieves this by having parser combinators automatically generate errors based on their parser function's type signature while providing the [`Parser::map_err`] utility to convert the original error type into your own.
```

Comment:
Hmm, I think this paragraph feels a bit fluffy and doesn't really say much. May I recommend:

> Chumsky manages the tracking and generation of errors for you. However, you may customise what information an error keeps track of and how it tracks it by changing the error type (see [`extra::Err`]). Chumsky provides a range of built-in error types to handle a range of use-cases, but you can also implement the [`Error`] trait for your own error type if you want even more fine-grained control.

In my view, talking about `map_err` is probably a point for another heading - it, along with combinators like `try_map`, `validate`, etc. are more in the realm of error manipulation rather than actually defining properties of errors.